frontend/packages/core/src/Contexts/workflow-storage-context/reducer.tsx

```diff
@@ -0,0 +1,74 @@
+import { removeLocalData, rotateDataFromAPI, storeLocalData } from "./helpers";
+import type { Action, BackgroundPayload, UserPayload, WorkflowStorageState } from "./types";
+
+/**
+ * Reducer for the WorkflowStorageContext
+ * This will act on the WorkflowStorageState and add / remove items from the temporary storage
+ * as well as localStorage, this will (optionally) keep all storage actions in one location for all
+ * workflows and allow for easier state hydration
+ */
+const workflowStorageContextReducer = (
+  state: WorkflowStorageState,
+  action: Action
+): WorkflowStorageState => {
+  switch (action.type) {
+    // Will add data to our temporary storage as well as the local storage
```

Comment:
We might have already discussed this but just to refresh myself, is there a reason to add this to local storage and temporary?

Reply:
We synced on this but to make it transparent:

Example:

"ProjectSelector will save its state, which will write to temp and then it will forward it to local storage (since it needs local storage for its own use cases). So the temp storage could be used if you decide to create a short link at that moment in time and then you could make further changes and reload and it would pull from your local storage as normal."

"If you just want to be able to hydrate the state for a short link but not save anything in local storage you would just use the store the exact same but not pass the local storage option."